server/py/services/api/crud/secrets.py

```diff
@@ -596,7 +598,21 @@ def revoke_secret_token(
         )
 
     @staticmethod
-    def _validate_token_name(token_name: str, seen_names: set):
+    def _validate_token_name_and_user(
+        secret_token: mlrun.common.schemas.SecretToken,
+        seen_names: set,
+        authenticated_id: str,
+    ):
+        token_name = secret_token.name
+        token_sub = Secrets._decode_offline_token(token_name, secret_token.token).get(
+            "sub"
+        )
+
+        if token_sub != authenticated_id:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Offline token '{token_name}' subject '{token_sub}' does not match the authenticated user ID. "
+                "Stored tokens can only belong to the authenticated user."
```

Comment:
I dont think that revealing the subject makes sense from client POV. you may log it with "warning" and raise the exception

Reply:
When you say log with "warning" and then raise the exception, do you mean like this:

```suggestion
            if token_sub != authenticated_id:
               mlrun.utils.logger.warning(f"Offline token '{token_name}' subject {token_sub} does not match the authenticated user ID: {authenticated_id}.")
                raise mlrun.errors.MLRunInvalidArgumentError(
                    f"Offline token '{token_name}' does not match the authenticated user ID. "
                    "Stored tokens can only belong to the authenticated user."
            )
```

Reply:
@liranbg Please do not commit this suggestion as it will require changes in some tests

Reply:
@liranbg I've committed this change for now. Let me know if you think it needs to be different